### PR TITLE
Insert *~ when using emacs in .gitignore

### PR DESCRIPTION
**Pull request by truongnh1992**
Ignore temporary files *~ when editting source codes with
emacs.

Signed-off-by: Nguyen Hai Truong <truongnh@vn.fujitsu.com>

For an explanation of how to fill out the fields, please see the relevant section 
in [PULL_REQUESTS.md](./PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by mattklein123** [comment]
Needs master merge. In the future please open all of these 1-liners in a single PR. Thanks.

/wait

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,4 @@ TAGS
 .vs
 .vscode
 .idea/
+*~
```
Comment (jmarantz) [inline .gitignore]: I use Emacs and have plenty of files lying around like ./test/common/stats/stats_speed_test.cc.~1~ and they don't cause any trouble. I was wondering if this was some general setting I had, but I can't find it.

So my opinion is that git does the right thing for emacs old-version files by default and this is not necessary.

**Comment by mattklein123** [comment]
@jmarantz SGTM we can close.

**Comment by jmarantz** [comment]
Actually my bad...the reason this works for me is because I have a process for creating new clients that involves me manually editing `./.git/info/exclude` to uncomment the last two lines:
```
*.[oa]
*~
```

Not having to do this for each new client would be great.  So can we add `*.[oa]` to this PR after merging master?

**Pull request closed without merging**